pkg/server/router/common.go

```diff
@@ -31,25 +31,40 @@ import (
 // to Router and Gateway.
 type commonRouter struct {
 	payloadFunc      func(req *http.Request) (p *Payload, err error)
-	matchHandlerFunc func(req *http.Request, p *Payload) (h Handler, pp []Processor)
+	matchHandlerFunc func(path string, method string, p *Payload) (h Handler, pp []Processor)
```

Comment:
The payload design rationale is: payload contain all information on the request. Transport may duplicate the information to transport specific place. For example, HTTP path in HTTP transport, access_token at HTTP header. But the payload will always contains all information skygear server need to process the request. The additional copy will assumed to be for fast routing.

In the case of ZMQ, we may copy the action into routing frame. It will enable the program to route the request without actually parsing the whole JSON.